app/scripts/lib/errors.js

```diff
@@ -116,10 +120,9 @@ define(function (require, exports, module) {
      * @returns {Error}
      */
     toError (type, context) {
-      var errno = this.toErrno(type);
-      var message = this.toMessage(errno);
-
-      var err = new Error(message);
+      const errno = this.toErrno(type);
+      const message = this.toMessage(errno);
+      const err = _.isString(message) ? new Error(message) : message;
```

Comment:
`new Error(message) : message;` if we just get a plain `message` does this mean this is already an error? 

or should we check if it is not already `Error` and then create `new Error`?

Reply:
Good thinking, I'll do that.

Reply:
Or maybe I won't. That would still give us potential problems if the result was a non-`Error` object type.

Reply:
(and actually, `new Error(new Error('foo'))` is safe to do)